Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "linemux"
+version = "0.0.1"
+authors = ["Jon Magnuson <jon.magnuson@gmail.com>"]
+edition = "2018"
+description = "A library providing asynchronous, multiplexed tailing for (namely log) files."
+documentation = "https://docs.rs/linemux"
+repository = "https://github.com/jmagnuson/linemux"
+readme = "README.md"
+keywords = ["tail", "lines", "watch", "fs", "events"]
+license = "MIT OR Apache-2.0"
+categories = ["asynchronous", "filesystem"]
+
+[dependencies]
+async-trait = "0.1"
+futures-util = "0.3"
+notify = "5.0.0-pre.2"
+pin-project = "0.4"
```

Comment:
Are you using specific features from `pin-project`, or would `pin-project-lite` be more appropriate?

Reply:
I had noticed that destructuring the `Pin` was a common pattern among components dealing with multiple Futures/Streams (e.g., [`futures_util::stream::Select`](https://github.com/rust-lang/futures-rs/blob/d98e6ecb231988ba0d017870ee95ca51a2c7315c/futures-util/src/stream/select.rs#L61)), and figured I would want to do similarly to avoid any potential ownership issues while a la carte'ing field references. [Only `pin-project` supports destructuring](https://github.com/taiki-e/pin-project-lite#different-no-support-for-pattern-matching-and-destructing), but I'll do a quick pass with `pin-project-lite` to see if I can make it work in the state machine. It should speed up compile times, so that would be nice.